### ISSUE TITLE
performance for stage2

### ISSUE THREAD

**Issue by xup16** [huochaitiantang/pytorch-deep-image-matting#33]
Hi~ Thank you for the excellent work.
I have reproduced the performance of stage1 followed you codes, but I can not reproduce the performance of stage2 in the paper (50 SAD).
Would you provide your performance and model of stage2 if you have tried.
Thanks!

**Comment by huochaitiantang** [comment]
I tried stage2 training(resume from sad=54.42 model and only train the convolutions of the refine stage), but the performance is not as good as paper(our best sad=53.74). There may be some mistakes in the statge2 training codes(loss function or network structure).

**Comment by xup16** [comment]
Thank you for the reply. 
Did you try stage3？Training the encoder,  decoder and refine stage end to end？

**Comment by huochaitiantang** [comment]
Yes, I also tried the stage3 training(resume from stage2 sad=53.74 and train the whole network end-to-end) but got the worse performance(best sad= 55.48). There must exist some mistakes in the refine stage training. Maybe you could help check the implementation code.

**Comment by xup16** [comment]
Ok. Thank you very much

**Comment by AstonyJ** [comment]
Hi~ Thank you for the excellent work.
1.I have trained stage1 from scratch followed you codes, but I get 59.40.I set the lr 0.00001 constantly,but i see your code would adjust the lr.Is this the reason？
2.How do you train stage 2？I tried, but the effect was bad.(resume from stage1 59.40,batch_size=4,4cards).Can you show me the parameter set in stage2? 
Thank you.

**Comment by wrrJasmine** [comment]
@huochaitiantang I also met the problem，I found that the composition loss is really hard to train. have you found the reason?

**Comment by SahadevPoudel** [comment]
> Hi~ Thank you for the excellent work.
> 1.I have trained stage1 from scratch followed you codes, but I get 59.40.I set the lr 0.00001 constantly,but i see your code would adjust the lr.Is this the reason？
> 2.How do you train stage 2？I tried, but the effect was bad.(resume from stage1 59.40,batch_size=4,4cards).Can you show me the parameter set in stage2?
> Thank you.

Hi, i have trained stage 1 from scratch and i run same code. but i get 86.77. did u make any changes?
